ee/server/service/maintained_apps.go

```diff
@@ -200,6 +202,29 @@ func (svc *Service) AddFleetMaintainedApp(
 		return 0, ctxerr.Wrap(ctx, err, "creating activity for added software")
 	}
 
+	if automaticInstall {
+		policies, err := svc.ds.GetPoliciesBySoftwareTitleIDs(ctx, []uint{titleID}, teamID)
+		if err != nil {
+			return 0, ctxerr.Wrap(ctx, err, "getting new automatic install policy for FMA")
+		}
+
+		if len(policies) == 0 {
+			// Something very weird has happened if we don't find
```

Comment:
We need to read from the writer to avoid this. Or pass the policy ID back when creating it so we have the pieces we need to create the activity without going back to the DB.